core/common/src/main/java/alluxio/exception/ExceptionMessage.java

```diff
@@ -81,6 +84,7 @@ public enum ExceptionMessage {
   CANNOT_FREE_PINNED_FILE("Cannot free file {0} which is pinned. Please unpin it first or"
       + " set the \"forced\" flag of free operation to true"),
   INODE_DOES_NOT_EXIST("inodeId {0,number,#} does not exist"),
+  PREFIX_DOES_NOT_MATCH_PATH("Prefix component {0} does not match path component {1}"),
```

Comment:
This will happen if a user does a batch listing that also uses a prefix, and in-between calls they change the prefix, so the prefix no longer matches the listing. So it should be a user error, I updated the message slightly that maybe makes it more clear.